VideoLocker/src/main/java/org/edx/mobile/util/UrlUtil.java

```diff
@@ -0,0 +1,26 @@
+package org.edx.mobile.util;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+/**
+ * Created by aleffert on 3/31/16.
+ */
+public class UrlUtil {
+
+    // Resolves a URL against a base URL. If the url is already absolute,
+    // it just returns it. If the URL is relative it will resolve it against the base URL
+    public static String makeAbsolute(String url, String base) {
+        if (url == null || base == null) {
+            return null;
+        }
+        try {
+            URI baseUri = new URI(base);
+            URI result = baseUri.resolve(url);
```

Comment:
It would be better to pass a new `URI` here, since the implementation of the `resolve()` method which takes a `String` parameter just calls through to `resolve(create(relative))`, and the `create()` method is designed to construct the `URI` while swallowing the checked `URISyntaxException` and throwing the unchecked `IllegalArgumentException` in it's stead. You might also want to construct a separate test where this method is called with just the relative URL malformed, as that would have exposed this issue.